### PR TITLE
Add support for new fee model

### PR DESCRIPTION
**Pull request by shresthagrawal**
Resolves BACK-378

### CODE REVIEW

**Review by ColonelJ — CHANGES_REQUESTED** [review]
There is a router for `buy` as well, not sure if you pulled it before starting this.

File: src/router/payload-encoder.ts [hunk 1]
```diff
@@ -12,6 +12,23 @@ import { SwapSide } from '../constants';
...
+) {
+  let fee = BigInt(partnerFeePercent);
+  if (fee > 10000) throw new Error('fee bps should be less than 10000');
+
+  // Set 14th bit if positiveSlippageToUser is true
+  if (positiveSlippageToUser) fee = fee | (BigInt(1) << BigInt(14));
+
+  // Bits 248 - 255 is used for version;
+  // Set version = 1;
+  fee = fee | (BigInt(1) << BigInt(248));
```
Comment (ColonelJ) [inline src/router/payload-encoder.ts:27]: You can use `|=` on these two lines.

File: src/router/payload-encoder.ts [hunk 2]
```diff
@@ -12,6 +12,23 @@ import { SwapSide } from '../constants';
...
 
+export function encodeFeePercent(
+  partnerFeePercent: string,
+  positiveSlippageToUser: boolean,
+) {
+  let fee = BigInt(partnerFeePercent);
+  if (fee > 10000) throw new Error('fee bps should be less than 10000');
+
+  // Set 14th bit if positiveSlippageToUser is true
+  if (positiveSlippageToUser) fee = fee | (BigInt(1) << BigInt(14));
```
Comment (erez-paraswap) [inline src/router/payload-encoder.ts]: I'd recommend calculating (BigInt(1) << BigInt(14) once and the using the value as much as needed

File: src/router/payload-encoder.ts (hunk 1 above)
Comment (erez-paraswap) [inline src/router/payload-encoder.ts]: Same as above here

**Review by erez-paraswap — APPROVED** [review]
These are not functional problems, but we should definitely put more attention to code performance as we learn on our own skin

File: src/router/payload-encoder.ts (hunk 2 above)
Comment (shresthagrawal) [reply src/router/payload-encoder.ts]: fixed

File: src/router/payload-encoder.ts (hunk 1 above)
Comment (shresthagrawal) [reply src/router/payload-encoder.ts]: fixed

**Pull request merged**